File: stack/php/pairedParentheses.php
<?php

declare(strict_types=1);

function pairedParentheses(string $s): bool
{
    $balance = 0;
    foreach (str_split($s) as $ch) {
        if ($ch === '(') {
            $balance++;
        } elseif ($ch === ')') {
            $balance--;
            if ($balance < 0) {
                return false;
            }
        }
    }
    return $balance === 0;
}

File: stack/csharp/PairedParentheses.cs
namespace StackSolutions;

public static class PairedParentheses
{
    public static bool Check(string s)
    {
        var balance = 0;
        foreach (var ch in s)
        {
            if (ch == '(')
            {
                balance++;
            }
            else if (ch == ')')
            {
                balance--;
                if (balance < 0)
                {
                    return false;
                }
            }
        }

        return balance == 0;
    }
}
